### PR TITLE
[tests] Doctests

### PR DESCRIPTION
**Pull request by stevhliu**
Start to add doctests for code examples in the Diffusers docs :)

todo:

- [x] schedule doctest to run on GitHub ([.github/workflows/doctests.yml](https://github.com/huggingface/transformers/blob/b074461ef0f54ce37c5239d30ee960ece28d11ec/.github/workflows/doctests.yml#L22))

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,6 +70,7 @@ fixup: modified_only_fixup extra_style_checks autogenerate_code repo-consistency
 fix-copies:
 	python utils/check_copies.py --fix_and_overwrite
 	python utils/check_dummies.py --fix_and_overwrite
+	python utils/check_doctest_list --fix_and_overwrite
```
Comment (DN6) [inline Makefile:73]:
```suggestion
	python utils/check_doctest_list.py --fix_and_overwrite
```

File: .github/workflows/doctests.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  schedule:
+    - cron: "17 2 * * *"
+
+
+env:
+  HF_HOME: /mnt/cache
+  RUN_SLOW: yes
+  OMP_NUM_THREADS: 16
+  MKL_NUM_THREADS: 16
+  SIGOPT_API_TOKEN: ${{ secrets.SIGOPT_API_TOKEN }}
```
Comment (DN6) [inline .github/workflows/doctests.yml:17]: Is this for https://sigopt.com/ ? Any reason we need it?

Comment (stevhliu) [reply .github/workflows/doctests.yml]: Removed since not relevant to Diffusers :)

**Comment by yiyixuxu** [comment]
we are still working on this no?

**Comment by yiyixuxu** [comment]
@stevhliu 
we can close this one now that we have #7145?

**Comment by stevhliu** [comment]
Sounds good to me 🙂

**Pull request closed without merging**